runtime/src/lib.rs

```diff
@@ -544,11 +548,25 @@ impl TrieVerifier for ProofTrieVerifier {
     }
 }
 
+type ThresholdType = i128;
+
+pub struct ThresholdDecayFunction;
+impl Get<ThresholdType> for ThresholdDecayFunction {
+    fn get() -> ThresholdType {
```

Comment:
Wouldn't this yield `0`? We're returning an integer division whose result is smaller than `1`

Reply:
I am not sure I understand how it would yield `0`... Why can't it return `0.5`?

It's 128-bit **signed** integer type.